app/pages/OrganizationDetailPage.tsx

```diff
@@ -35,18 +35,62 @@ export const OrganizationDetailPage = () => {
   const searchState = useMemo(() => qs.parse(search.slice(1)), [search]);
   const { visitDeactivated } = searchState;
 
-  useEffect(() => window.scrollTo(0, 0), []);
+  const [error, setError] = useState<Error | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
 
+  useEffect(() => window.scrollTo(0, 0), []);
   useEffect(() => {
-    fetchOrganization(organizationListingId as string).then((o) => setOrg(o));
-    // TODO Handle Errors
+    const fetchOrganizationWithErrorHandling = async () => {
+      setIsLoading(true);
+      try {
+        const org = await fetchOrganization(organizationListingId as string);
+
+        if ("message" in org) {
+          throw new Error();
+        }
```

Comment:
**issue:** @adriencyberspace I'm not sure this check will work as you might think? The `fetchOrganization` function does not return a message property. Were you looking at `fetchService`? You may need to replicate the response handling and return values I added there a few months ago for this check to work similarly.

Reply:
I don't have time to check the app this sec but I can perhaps a bit later or tomorrow.

Reply:
Oops! I was referencing the ServiceDetailPage logic and then didn't check that part. Removed it

Reply:
Should I add error handling logic to `Organization.ts` instead?